tracing-log/src/lib.rs

```diff
@@ -246,37 +258,14 @@ pub struct TraceLoggerBuilder {
 
 // ===== impl LogTracer =====
 
-impl LogTracer {
-    pub fn with_filter(filter: log::LevelFilter) -> Self {
-        Self { filter }
-    }
-}
-
-impl Default for LogTracer {
-    fn default() -> Self {
-        Self::with_filter(log::LevelFilter::Info)
-    }
-}
-
 impl log::Log for LogTracer {
-    fn enabled(&self, metadata: &log::Metadata) -> bool {
-        metadata.level() <= self.filter
+    fn enabled(&self, _metadata: &log::Metadata) -> bool {
+        // Use log::set_max_level to filter LogTracing's input
+        true
     }
 
     fn log(&self, record: &log::Record) {
-        let enabled = dispatcher::get_default(|dispatch| {
-            // TODO: can we cache this for each log record, so we can get
-            // similar to the callsite cache?
-            dispatch.enabled(&record.as_trace())
-        });
-
-        if enabled {
-            // TODO: if the record is enabled, we'll get the current dispatcher
-            // twice --- once to check if enabled, and again to dispatch the event.
-            // If we could construct events without dispatching them, we could
-            // re-use the dispatcher reference...
-            format_trace(record).unwrap();
-        }
+        format_trace(record).unwrap();
```

Comment:
This should still check if the record is enabled by the default dispatcher. Otherwise, the trace dispatcher will not be able to filter the log event. I think this code should not have been deleted.

Reply:
Definitely